### PR TITLE
DEVPROD-11166 Add general documentation for Evergreen task statuses 

### PR DESCRIPTION
**Pull request by khelif96**
DEVPROD-11166

### Description
Adds documentation for all Evergreen task statuses and display task statuses. This documentation should be easily searchable in pine and provides an overview of how we calculate task statuses

Reference:
https://github.com/evergreen-ci/evergreen/blob/64cc9b4b15f527537da6c998582104f0a5fabe79/globals.go#L49
https://github.com/evergreen-ci/evergreen/blob/85529ffcf3e19800f49dc80a4b801ba1f01bd230/model/task/db.go#L204

### Documentation
This is a documentation change

### CODE REVIEW

File: docs/Reference/Task-Statuses.md [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+This document provides an overview of the various task statuses used in our
+system, along with their meanings, usage contexts, and the logic that determines
+these statuses.
+
+---
+
+## Table of Contents
+
+- [Active Task Statuses](#active-task-statuses)
```
Comment (hadjri) [inline docs/Reference/Task-Statuses.md:11]: Undispatched can be inactive, so I'd recommend saying something like "Unfinished task statuses" or move undispatched to its own category.

File: docs/Reference/Task-Statuses.md [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+### `failed`
+
+**Description**: The task has finished but encountered failures. This status
+covers any failure reason, which can be detailed in the task's end details.
+
+**Usage**: General failure status for tasks that did not complete successfully.
+
+---
+
+### `aborted`
```
Comment (hadjri) [inline docs/Reference/Task-Statuses.md:84]: Aborted is actually not a valid DB status field, only a display status for tasks that have `t.Aborted == true`. The only values the status field can have in DB for a finished task are `success` or `failed`.

File: docs/Reference/Task-Statuses.md [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+**Description**: The task is scheduled to run eventually, assuming all
+dependencies are met (`Activated == true` and `Status == "undispatched"`).
+
+**Usage**: Indicates tasks in the queue that are pending execution upon
+dependency resolution.
+
+---
+
+### `pending`
```
Comment (hadjri) [inline docs/Reference/Task-Statuses.md:273]: I have never seen this, does this exist?

File: docs/Reference/Task-Statuses.md [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+---
+
+## Final Task Statuses
+
+These statuses indicate that a task has completed its execution.
+
+### `success`
+
+**Description**: The task has finished successfully without any errors.
```
Comment (hadjri) [inline docs/Reference/Task-Statuses.md:69]: There is an asterisk here where a task can have a success status even if there are errors in the `post` block, and `post_error_fails_task: false`

**Review by hadjri — CHANGES_REQUESTED** [review]
This is great, thanks for doing this!

File: docs/Reference/Task-Statuses.md (hunk 3 above)
Comment (khelif96) [reply docs/Reference/Task-Statuses.md:273]: This is the state a task dependency can be in but not the task itself. https://github.com/evergreen-ci/evergreen/blob/4394a9d14c4c485f76e0af36fd2a5d2ed3a88ed6/service/task.go#L452

**Review by hadjri — APPROVED** [review]

**Pull request merged**